checkov/common/checks_infra/solvers/attribute_solvers/equals_attribute_solver.py

```diff
@@ -13,6 +13,11 @@ def __init__(self, resource_types: List[str], attribute: Optional[str], value: A
 
     def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:
         attr_val = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
+        # if this value contains an underendered variable, then we cannot evaluate the check,
+        # so return True (since we cannot return UNKNOWN)
+        # handle edge cases in some policies that explicitly look for blank values
+        if self.value != '' and self._is_variable_dependant(attr_val, vertex['source_']):
```

Comment:
@mikeurbanski1 wasn't it possible to implement this once in `get_operation` in `base_attribute_solver`?

Reply:
I think it only applies to `equals`. I added it because a single OOTB policy test was failing. The logic for the policy is `attribute_1 equals ''`. Thus, if the resource has `attribute_1 = aws_instance.i.id`, for example, this check was automatically returning `True`, when of course any value for the variable would make it fail.

There are many other possible little cases like this that could arise for specific policies and operators, so I decided to just solve this one policy for now.